gunrock/oprtr/intersection/kernel.cuh

```diff
@@ -182,40 +182,38 @@ cudaError_t Launch(gunrock::util::MultiGpuContext mgpu_context,
   state.Save();
 
   size_t input_length = graph.edges;
-  size_t stride =
-      (input_length + KernelPolicyT::BLOCKS * KernelPolicyT::THREADS - 1) >>
-      (KernelPolicyT::LOG_THREADS + KernelPolicyT::LOG_BLOCKS);
-
   SizeT num_vertices = graph.nodes;
 
-  SizeT edges_per_gpu = gunrock::util::ceil_divide(input_length, mgpu_context.getGpuCount());
-
   printf("Num verts %d\n", num_vertices);
-  printf("edges_per_gpu %d\n", edges_per_gpu);
 
   std::vector<std::thread> threads;
   threads.reserve(mgpu_context.getGpuCount());
 
-  for (int i = 0; i < mgpu_context.getGpuCount(); ++i) {
-    auto context = mgpu_context.contexts[i];
-    auto rows_offset = edges_per_gpu * i + graph.CsrT::row_offsets.GetPointer(util::DEVICE);
-    auto cols_offset = edges_per_gpu * i + graph.CsrT::column_indices.GetPointer(util::DEVICE);
-
-    printf("Rows offset %p\n", rows_offset);
-    printf("Cols offset %p\n", cols_offset);
+  for (auto& context : mgpu_context.contexts) {
 
     threads.push_back(std::thread( [&, context]() {
-        
+        SizeT edges_per_gpu = gunrock::util::ceil_divide(input_length, mgpu_context.getGpuCount());
+
```

Comment:
I believe at the moment this only needs to happen once and can be moved outside of the range-based for loop. 

That being said, there could be a future where we're smarter about how many `edges_per_gpu` we dole out (doesn't necessarily have to be the same number for every GPU and could be based on some sort of "load" information). Let's explore that when it becomes an issue.